pkg/apiserver/handlers/featuregates/handler.go

```diff
@@ -48,80 +46,106 @@ type (
 	return func(w http.ResponseWriter, r *http.Request) {
 		antreaConfigName := env.GetAntreaConfigMapName()
 		antreaConfig, err := k8sclient.CoreV1().ConfigMaps(env.GetAntreaNamespace()).Get(context.TODO(), antreaConfigName, metav1.GetOptions{})
 		if err != nil {
 			w.WriteHeader(http.StatusInternalServerError)
-			klog.Errorf("Error when getting config map %s: %v", antreaConfigName, err)
+			klog.ErrorS(err, "Error when getting config map", antreaConfigName)
+			return
+		}
+		configMaps, err := k8sclient.CoreV1().ConfigMaps(env.GetAntreaNamespace()).List(context.TODO(), metav1.ListOptions{
+			LabelSelector: "app=antrea",
+		})
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			klog.ErrorS(err, "Error when getting antrea-windows config map")
 			return
 		}
+		antreaWindowsConfigMaps := []v1.ConfigMap{}
+		for _, cm := range configMaps.Items {
+			if strings.HasPrefix(cm.Name, "antrea-windows-config") {
+				antreaWindowsConfigMaps = append(antreaWindowsConfigMaps, cm)
+			}
+		}
 
 		agentConfig := &Config{}
-		err = yaml.Unmarshal([]byte(antreaConfig.Data["antrea-agent.conf"]), agentConfig)
+		err = yaml.Unmarshal([]byte(antreaConfig.Data[agentConfigName]), agentConfig)
 		if err != nil {
 			w.WriteHeader(http.StatusInternalServerError)
-			klog.Errorf("Failed to unmarshal Antrea antrea-agent.conf: %v", err)
+			klog.ErrorS(err, "Failed to unmarshal Antrea", agentConfigName)
 			return
 		}
 
-		agentfeatureGates := getAgentGatesResponse(agentConfig)
-		controllerfeatureGates := getControllerGatesResponse()
-		result := append(agentfeatureGates, controllerfeatureGates...)
-		err = json.NewEncoder(w).Encode(result)
+		controllerConfig := &Config{}
+		err = yaml.Unmarshal([]byte(antreaConfig.Data[controllerConfigName]), controllerConfig)
 		if err != nil {
 			w.WriteHeader(http.StatusInternalServerError)
-			klog.Errorf("Error when encoding FeatureGates to json: %v", err)
+			klog.ErrorS(err, "Failed to unmarshal Antrea", controllerConfigName)
 			return
 		}
-	}
-}
 
-func getAgentGatesResponse(cfg *Config) []Response {
-	gatesResp := []Response{}
-	for df := range features.DefaultAntreaFeatureGates {
-		dfs := string(df)
-		if !agentGates.Has(dfs) {
-			continue
+		agentfeatureGates := getFeatureGatesResponse(agentConfig, agentMode)
+		controllerfeatureGates := getFeatureGatesResponse(controllerConfig, controllerMode)
+		result := append(agentfeatureGates, controllerfeatureGates...)
+
+		if len(antreaWindowsConfigMaps) > 0 {
+			sort.Slice(antreaWindowsConfigMaps, func(i, j int) bool {
+				return antreaWindowsConfigMaps[i].CreationTimestamp.After(antreaWindowsConfigMaps[j].CreationTimestamp.Time)
+			})
+			agentWindowsConfig := &Config{}
+			err = yaml.Unmarshal([]byte(configMaps.Items[0].Data[agentConfigName]), agentWindowsConfig)
+			if err != nil {
+				w.WriteHeader(http.StatusInternalServerError)
+				klog.ErrorS(err, "Failed to unmarshal Antrea windows", agentConfigName)
```

Comment:
error log is inaccurate, it doesn't mention that we are talking about a 
"ConfigMap"

the call is also invalid (see above), since there is no log key for the `agentConfigName` value